test/battle/move_effect/multi_hit.c

```diff
@@ -183,3 +183,51 @@ SINGLE_BATTLE_TEST("Endure does not prevent multiple hits and stat changes occur
         MESSAGE("Wobbuffet's Speed rose!");
     }
 }
+
+SINGLE_BATTLE_TEST("Scale Shot decreases defense and increases speed after the 4th hit of Loaded Dice")
+{
+    PASSES_RANDOMLY(50, 100, RNG_LOADED_DICE);
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_SCALE_SHOT].effect == EFFECT_MULTI_HIT);
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_LOADED_DICE); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_SCALE_SHOT); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_SCALE_SHOT, player);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_SCALE_SHOT, player);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_SCALE_SHOT, player);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_SCALE_SHOT, player);
+        MESSAGE("Hit 5 time(s)!");
```

Comment:
If it's hitting 5 times, why is it only playing the animation 4 times?

Reply:
Good question. It should be 4 times. It passes because it can randomly pass with 5 hits.